examples/linux_asyncio_consumer.py

```diff
@@ -0,0 +1,164 @@
+            AsyncConsumer.close_eventd, self.loop, self.eventfd
+        )
+
+    @staticmethod
+    def close_eventd(loop, eventfd):
+        loop.remove_reader(eventfd)
+        os.close(eventfd)
+
+    def close(self):
+        self.consumer.close()
+
+    def __eventfd_ready(self):
+        os.eventfd_read(self.eventfd)
+
+        for future in self.waiters:
+            if not future.done():
+                future.set_result(True)
+
+    def subscribe(self, *args, **kwargs):
+        self.consumer.subscribe(*args, **kwargs)
+
+    def assign(self, *args, **kwargs):
+        self.consumer.assign(*args, **kwargs)
+
+    async def poll(self, timeout=0):
+        if timeout > 0:
+            try:
+                return await asyncio.wait_for(self._poll_no_timeout(), timeout)
+            except asyncio.TimeoutError:
+                return None
+        else:
+            return self._poll_no_timeout()
+
+    async def _poll_no_timeout(self):
+        while not (msg := await self._single_poll()):
+            pass
+        return msg
+
+    async def _single_poll(self):
+        if (msg := self.consumer.poll(timeout=0)) is not None:
+            return msg
+
+        awaitable = self.loop.create_future()
+        self.waiters.add(awaitable)
+        try:
+            # timeout=2 is there for two reasons:
+            # 1) self.consumer.poll needs to be called reguraly for other
+            #    activities like: log callbacks.
+            # 2) Ensures progress even if something with eventfd
+            #    notification goes wrong.
+            await asyncio.wait_for(awaitable, timeout=2)
+        except asyncio.TimeoutError:
+            return None
+        finally:
+            self.waiters.discard(awaitable)
+
+
+async def main():
+    arguments = parse_args()
+
+    logger = logging.getLogger('consumer')
+    logger.setLevel(logging.DEBUG)
+    handler = logging.StreamHandler()
+    fmt = '%(asctime)-15s %(levelname)-8s %(message)s'
+    handler.setFormatter(logging.Formatter(fmt))
+    logger.addHandler(handler)
+
+    config = {
+            'bootstrap.servers': arguments.bootstrap_servers,
+            'group.id': arguments.group_id,
+            'session.timeout.ms': 6000,
+            'auto.offset.reset': 'earliest'
+    }
+
+    if arguments.client_status:
+        config['stats_cb'] = stats_cb
+        config['statistics.interval.ms'] = arguments.client_status
+
+    consumer = AsyncConsumer(config, logger)
+    consumer.subscribe(arguments.topic, on_assign=print_assignment)
+
+    try:
+        while True:
+            msg = await consumer.poll(timeout=10)
+            if msg is None:
+                continue
+
+            if msg.error():
+                print(KafkaException(msg.error()))
+            else:
+                print('%% %s [%d] at offset %d with key %s:\n' % (
+                    msg.topic(),
+                    msg.partition(),
+                    msg.offset(),
+                    str(msg.key())
```

Comment:
this is likely to fail if the key is not indeed a string, recommend not printing the key at all, or printing as hex

Reply:
As far as I can see python3 tries to string format `bytes` already as nice as possible. Take for example:
```
>>> key = b'HELLO-\x00\x18'
>>> len(key)
8
>>> print('%s' % key)
b'HELLO-\x00\x18'
```
So it kept it nearly as it is, but just removed the `str(..)`, because the format already happens via the `%s`.